src/components/Table/index.tsx

```diff
@@ -155,9 +161,15 @@ class Table<T> extends React.Component<TableProps<T>, TableState<T>> {
 		};
 	}
 
-	componentWillReceiveProps(newProps: TableProps<T>) {
-		if (newProps.sort !== this.props.sort) {
-			this.setState({ sort: newProps.sort } as TableState<T>);
+	// trick for static method with generic: https://github.com/Microsoft/TypeScript/issues/24018
+	static getDerivedStateFromProps<S extends TableField>(
```

Comment:
All you need to do is change this to:
```
	static getDerivedStateFromProps<T>(
		nextProps: TableProps<T>,
		prevState: TableProps<T>,
	) {
		if (nextProps.sort !== prevState.sort) {
			return { sort: {...nextProps.sort} };
		} else {
			return null;
		}
	}
```

and get rid of the additional type you defined above.